### ISSUE TITLE
"ResizeObserver loop completed with undelivered notifications" makes Jasmine fail with empty error message

### ISSUE THREAD

**Issue by ej612** [jasmine/jasmine#2029]
### Your question

Hi there!

Recently, we're being increasingly plagued by flaky tests that fail with an empty error message:

![image](https://github.com/jasmine/jasmine/assets/110408767/f76a9d94-86ba-4af9-bb01-4717a3fd41ee)

By adding a breakpoint [here](https://github.com/jasmine/jasmine/blob/main/lib/jasmine-core/jasmine.js#L4068), I was able to find the culprit:

![image](https://github.com/jasmine/jasmine/assets/110408767/b554ccab-e07e-4f00-9940-89da232171f2)

I've researched this error message extensively and it looks as if it originates from one of our dependencies. So I need a way to suppress the error while they figure out what's going on. We've also never seen it in prod, only in Jasmine. To suppress it, I have overwritten `window.onerror`, I have done `addEventListener('error')` and called both `event.stopPropagation()` as well as `event.stopImmediatePropagation()`, to no avail.

It seems there's some monkey business going on, including the fact that Jasmine fails with an empty message instead of the error. I don't suppose the little information I can give you will be of much help to find the root cause, I'm just trying to find a way to ignore the error:
If I add a breakpoint [here](https://github.com/jasmine/jasmine/blob/main/lib/jasmine-core/jasmine.js#L1288), it gets called everytime I run into the error. Three lines above, there's an `if` condition:
```
if (!options.suppressLoadErrors) {
```
It looks as if I'll be able to suppress the error by enabling this option. Is there a way for me to do that?

Any help is appreciated. Thanks a lot in advance!

### Example code that demonstrates the problem

```JavaScript
N/A
```

### jasmine-core version

5.1.2

### Node.js or browser version

Node 18

### Operating System

Win11

**Comment by sgravrock** [comment]
This is very strange. Among other oddities, load-time errors are reported on the top suite, not on individual specs.

How are you running Jasmine? Are you using the standalone distribution, or have you installed some other package that loads and runs Jasmine? (e.g. jasmine-browser-runner, karma, webdriverio, etc.) Configuration is controlled by the tool you use to run Jasmine, so even the answer to "how do I set suppressLoadErrors" depends on that.

Is there anything in your stack that might be monkey patching Jasmine in any way? (e.g. zone.js, webdriverio, some code that somebody added to your codebase 5 years ago because they didn't like something about Jasmine, etc.)

**Comment by sgravrock** [comment]
Can you turn off stack trace filtering (change `if (frame.file !== jasmineFile)` to `if (true)`), run it again, and post the resulting reporter output?

**Comment by sgravrock** [comment]
> We're running Jasmine with karma.

In that case, check the Karma docs for how to configure Jasmine.

But I don't think turning off load-time error handling is going to help you. It looks like the error is happening repeatedly while the specs are running, not just at startup. I don't think you'll be able to get your suite green without actually stopping the error from happening. Although the lack of an error message is a problem, Jasmine is right to fail your tests when the global error event fires.

My best guess is that the library in question is manually triggering an `ErrorEvent` without including an actual error. That's technically valid (the `error` property is allowed to be literally anything, including null/undefined) but it's bad form because it rules out the possibility of a stack trace, and it's also a bit of a dusty corner of the error-handling space. I'm not entirely surprised that Jasmine has trouble with it. We could try to improve Jasmine's handling of this case but that wouldn't make your suite pass.

**Comment by sgravrock** [comment]
> I did that before opening this ticket, but couldn't find anything about this at all. Is suppressLoadErrors an option like random, seed, stopOnSpecFailure, specFilter etc?

No. It needs to be passed to the first call to `jasmine.getEnv()`. It's also the only configuration property that's passed that way, so if the Karma docs don't specifically mention `suppressLoadErrors` then there probably isn't a way to pass it.

In any case, `suppressLoadErrors` only makes Jasmine ignore errors that happen during script loading. It has no bearing on what happens later on, such as when specs are executing. 

> Is there a way I can overwrite Jasmine's default error handling? Otherwise we have no choice but to disable the affected tests.

If the errors only occur during the execution of particular specs, you could do something like this:

```
it('produces global errors that I want to ignore', async function() {
    // IMPORTANT: await the returned promise before returning
    // Leaving spyOnGlobalErrorsAsync installed at the end of an it/beforeEach/etc
    // is unsupported and pretty likely to break things. 
    await jasmine.spyOnGlobalErrorsAsync(async function(globalErrorSpy) {
        // Do the body of the spec
        // Finally, verify that globalErrorSpy was not called with anything other than the errors you expected
    }); 
});
```

**Comment by sgravrock** [comment]
That particular line of code is only used to dispatch errors to a `spyOnGlobalErrorsAsync` spy. It's not involved in sending errors to the HTML reporter.

In any case, the big piece of information that's missing here isn't what the problem with the code is. It's how to reproduce it. Without that, I can't assess whether it's worth the risk of changing the error handling code to address this and I can't tell whether a fix actually worked. If you'd like me to investigate any further, please provide a link to a repo containing a complete, working, reasonably minimal example project that reproduces the problem.

My best guess is still that you've got something that's dispatching an `ErrorEvent` without an actual error. If so, it's probably better to fix that problem at the source than to work around it in Jasmine. But it's tough to be sure without knowing what the source of the problem is.
